flake.nix

```diff
@@ -66,6 +69,12 @@
               # NOTE: we don't support nix gcc toolchains for darwin system, therefore explicitly setting CC is required
               pkgs.lib.optionalString pkgs.stdenv.isDarwin ''
                 export CC=gcc
+                export CXX=g++
+                for cmd in \
+                    ar as ld nm objcopy objdump readelf ranlib strip strings size windres
```

Comment:
Can you explain here and in the commit message what this is concretely needed for?

Reply:
We're using CC and LD now, and we might use AS in the future for building library, and some other env var might also be useful while developing. And since in here https://github.com/pq-code-package/mlkem-native/blob/d297d8639c821540fd735aae3018455e435b9bf6/flake.nix#L30

nix did this underlying
```
for cmd in \
    ar as ld nm objcopy objdump readelf ranlib strip strings size windres
do
    if
        PATH=$_PATH type -p "@targetPrefix@${cmd}" > /dev/null
    then
        export "${cmd^^}${role_post}=@targetPrefix@${cmd}";
    fi
done
```
https://github.com/NixOS/nixpkgs/blob/0c5b4ecbed5b155b705336aa96d878e55acd8685/pkgs/build-support/bintools-wrapper/setup-hook.sh#L57-L65

Therefore for system that installed toolchain from nix, these environment variables will be set properly. I'd like this to be consistent between different systems, and since we don't install toolchain from nix on macOS, we'll have to explicitly do this instead.